### DISCUSSION TITLE
Add Perplexity API

### DISCUSSION THREAD

**Discussion by Bigmanvst** [danielmiessler/Fabric#924] [Ideas]
Hi, 
Thanks a lot for this great tool, it change my mind of the AI considering the workflow in much more easy way. (within Obsidian plugin)

Can it be possible to integrate Perplexity API (openai structured ) :  https://docs.perplexity.ai/home ?
As a perplexity pro user I have 5$ included ("for free") for the API.
They have an huge model (405B) ❤️ : https://openrouter.ai/models/perplexity/llama-3.1-sonar-huge-128k-online

I can provide a perplexity API for testing, or I can perform some beta test.

Thanks a lot 
Regards

PS:
FYI -->  you can play with there models for free with this web : https://labs.perplexity.ai

**Comment by polytect** [comment]
This would be amazing!

**Comment by edward-walsh-gfk** [comment]
I would love to see this happen.

**Comment by anshuishere** [comment]
I was playing around a bit and found that Ppx API is OpenAI client-compatible.

I got it working with fabric using the following config (~/.config/fabric/.env)

OPENAI_API_KEY=pplx-apikey
OPENAI_API_BASE_URL=https://api.perplexity.ai
DEFAULT_VENDOR=OpenAI
DEFAULT_MODEL=sonar-pro

**Comment by LaBalaDeOro** [comment]
You can try out many different bots / models for free at https://poe.com

**Comment by ksylvan** [comment]
Done in #1527
